src/API/Google/Ads.php

```diff
@@ -323,4 +324,76 @@ private function get_merchant_link( int $merchant_id ): MerchantCenterLink {
 
 		throw new Exception( __( 'Merchant link is not available to accept', 'google-listings-and-ads' ) );
 	}
+
+	/**
+	 * Check if the user has accepted the customer data terms for enhanced conversion tracking.
+	 * Returns false for any account that fails.
+	 *
+	 * @return boolean
+	 */
+	public function get_accepted_customer_data_terms(): bool {
+		$ads_id = $this->options->get_ads_id();
+
+		try {
+			$accepted_terms = $this->options->get( OptionsInterface::ADS_CUSTOMER_DATA_TERMS, null );
+
+			// Retrieve the terms acceptance data from options.
+			if ( null !== $accepted_terms ) {
+				return (bool) apply_filters( 'woocommerce_gla_ads_enhanced_conversion_customer_data_terms', (bool) $accepted_terms );
+			}
+
+			$customer = ( new AdsAccountQuery() )
+				->set_client( $this->client, $ads_id )
```

Comment:
@mikkamp I think the issue was we were not checking whether the $ads_id is present or not. I have added that check at the top so that we can prevent the error.

Reply:
@mikkamp Sorry, it was not merged as the fix was added as part of a separate sub-PR created, but never merged because [that PR](https://github.com/woocommerce/google-listings-and-ads/pull/2316/files#diff-dcf740747692fe4bb7c17e8cd854c3b2d1fcdfb4392aa899eccc60d7363c9abcR334) was closed.

I have added the check and also tested that error should not appear after disconnecting the Google ads account.